master branch failed to build with LLVM 8

Is `julia` supposed to work with LLVM 8?  [Required Build Tools and External Libraries](https://github.com/JuliaLang/julia/blob/master/doc/build/build.md#required-build-tools-and-external-libraries) mentions that `julia` requires LLVM 6+.  However, it looks like building `julia` (105ed11b305b3208e37cf48f21812ef1e187d6f5) with LLVM 8 does not work:

```console
root@459e9488a96a:/julia# make -j12 USE_BINARYBUILDER_LLVM=0 BUILD_LLVM_CLANG=1 LLVM_VER=8.0.1
    CC src/jitlayers.o
/julia/src/jitlayers.cpp: In constructor 'JuliaOJIT::JuliaOJIT(llvm::TargetMachine&)':
/julia/src/jitlayers.cpp:477:9: error: 'AcknowledgeORCv1Deprecation' was not declared in this scope
         AcknowledgeORCv1Deprecation,
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:165: recipe for target 'jitlayers.o' failed
make[1]: *** [jitlayers.o] Error 1
Makefile:75: recipe for target 'julia-src-release' failed
make: *** [julia-src-release] Error 2
```

It works if I removed `AcknowledgeORCv1Deprecation`:

```diff
diff --git a/src/jitlayers.cpp b/src/jitlayers.cpp
index 603bb5b2e4..6368e5515f 100644
--- a/src/jitlayers.cpp
+++ b/src/jitlayers.cpp
@@ -474,7 +474,6 @@ JuliaOJIT::JuliaOJIT(TargetMachine &TM)
             cantFail(std::move(Err), "resolveSymbol failed");
           })),
     ObjectLayer(
-        AcknowledgeORCv1Deprecation,
         ES,
         [this](RTDyldObjHandleT) {
                       ObjLayerT::Resources result;
@@ -485,7 +484,6 @@ JuliaOJIT::JuliaOJIT(TargetMachine &TM)
         std::ref(registrar)
         ),
     CompileLayer(
-            AcknowledgeORCv1Deprecation,
             ObjectLayer,
             CompilerT(this)
         )
```

--- a/doc/build/build.md
+++ b/doc/build/build.md
@@ -166,7 +166,7 @@ downloaded (or in a few cases, included in the Julia source
 repository) and then compiled from source the first time you run
 `make`:
 
-- **[LLVM]** (6.0 + [patches](https://github.com/JuliaLang/julia/tree/master/deps/patches)) — compiler infrastructure (see [note below](#llvm)).
+- **[LLVM]** (9.0 + [patches](https://github.com/JuliaLang/julia/tree/master/deps/patches)) — compiler infrastructure (see [note below](#llvm)).
 - **[FemtoLisp]**            — packaged with Julia source, and used to implement the compiler front-end.
 - **[libuv]**  (custom fork) — portable, high-performance event-based I/O library.
 - **[OpenLibm]**             — portable libm library containing elementary math functions.

